### PR TITLE
Add labels API

### PR DESCRIPTION
**Pull request by davkal**
- `/api/v1/labels` to retrieve all seen label names
- `/api/v1/label/:name/values` to get all label values for a named label
- not bounded by time
- returns all labels in a list of strings in a `data` property

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -50,6 +56,51 @@ type Series struct {
...
+		http.Error(w, "Querier Error", http.StatusInternalServerError)
+	}
+	defer q.Close()
+
+	names, err := q.LabelNames()
+	if err != nil {
+		http.Error(w, "Querier Error", http.StatusInternalServerError)
+	}
+
+	res := &LabelResult{Labels: names}
```
Comment (brancz) [inline api/api.go]: Let’s rather create two distinct types for label names and label values.

**Comment by yeya24** [comment]
@davkal Hello, do you have plan to continue this pr? If not I am happy to work on it.

**Comment by bwplotka** [comment]
Happy to take this now

**Comment by bwplotka** [comment]
Will add series as well

**Comment by bwplotka** [comment]
-> https://github.com/conprof/conprof/pull/81

**Pull request closed without merging**